endtoend/helpers.go

```diff
@@ -0,0 +1,109 @@
+package endtoend
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func killProcesses(t *testing.T, pIDs []int) {
+	for _, id := range pIDs {
+		process, err := os.FindProcess(id)
+		if err != nil {
+			t.Fatalf("Could not find process %d: %v", id, err)
+		}
+		if err := process.Kill(); err != nil {
+			t.Fatal(err)
+		}
+		if err := process.Release(); err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func waitForTextInFile(file *os.File, text string) error {
+	wait := 0
+	// Cap the wait in case there are issues starting.
+	maxWait := 36
```

Comment:
Can we add a unit specified to this? Is it minutes/seconds? Maybe just use a deadline based channel: https://golang.org/pkg/context/#WithDeadline and doing a `for { select {` block that checks for `<-ctx.Done()`